src/main/java/org/openmrs/addonindex/service/ElasticSearchIndex.java

```diff
@@ -93,27 +93,39 @@ public void index(AddOnInfoAndVersions infoAndVersions) throws IOException {
 	}
 	
 	@Override
-	public Collection<AddOnInfoSummary> search(AddOnType type, String query) throws IOException {
+	public Collection<AddOnInfoSummary> search(AddOnType type, String query, String tag) throws IOException {
 		
 		BoolQueryBuilder boolQB = QueryBuilders.boolQuery();
 		if (type != null) {
-			//Gets all modules of the type specified by variable type
+			//Exact match on type
 			boolQB.filter(QueryBuilders.matchQuery("type", type));
 		}
+		if (tag != null) {
+			
+		SearchResult result = client.execute(new Search.Builder(new SearchSourceBuilder()
+				.size(SEARCH_SIZE)
+				.query(QueryBuilders.matchQuery("tags", tag)).toString())
+				.addIndex(AddOnInfoAndVersions.ES_INDEX)
+				.build());
+		return result.getHits(AddOnInfoAndVersions.class).stream()
```

Comment:
I think it's still overcomplicated.

The pseudocode should look like

    BoolQueryBuilder boolQB = QueryBuilders.boolQuery();
    if (type != null) {
        // add a clause to boolQB
    }
    if (tag != null) {
        // add a clause to boolQB
    }
    if (query != null) {
        // add lots of clauses to boolQB
    }
    // execute query and return results
You don't need to special-case `if (tag != null && query == null && type == null) {`